index.js

```diff
@@ -254,6 +254,15 @@ function buildRouteCompress (_fastify, params, routeOptions, decorateOnly) {
     if (payload == null) {
       return next()
     }
+
+    if (isFetchResponse(payload)) {
+      payload = payload.body
+    }
+
+    if (isWebReadableStream(payload)) {
+      payload = webStreamToNodeReadable(payload)
+    }
```

Comment:
I would move this code below, at the `if (typeof payload.pipe !== 'function') {`  level and write it as `if/else/if/else` or we are going to evaluate multiple times the same object